src/components/PopupWithConfirmation.js

```diff
@@ -0,0 +1,26 @@
+import Popup from './Popup.js';
+
+export default class PopupWithConfirmation extends Popup{
+    constructor({popupSelector, handleYesButton}){
+        super({popupSelector});
+        this._handleYesButton = handleYesButton;
+        this._popupButton = this._popupElement.querySelector('.modal__button');
+    }
+
+    setEventListeners(){
+        const deleteCard = (evt) => {
+            evt.preventDefault();
+            this._handleYesButton();
+            this._popupButton.removeEventListener('click', deleteCard);
+        }
+
+        const closePopup = () => {
+            this.close();
+            this._popupElement.querySelector('.modal__close-button-container').removeEventListener('click', closePopup);
+        }
+
+        this._popupElement.querySelector('.modal__close-button-container').addEventListener('click', closePopup);
+
```

Comment:
* Method `setEventListeners` inside `PopupWithConfirmation` is incorrect. Method `setEventListeners` should be called only **once** not to cause **memory leak**. And close icons are handled inside `Popup` already, so you can't handle it in`PopupWithConfirmation`


It should look something like that

```js

  setEventListeners() {
    super.setEventListeners();
    this._popupButton.addEventListener("click", () => {
      this._handleYesButton();
    });
  }
```


And changing the handler should be in `open` or in `setDeleteHandler`


```js
  setDeleteHandler(deleteCard) {
    this._handleYesButton = deleteCard
  }

```